src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -710,6 +710,9 @@ def python_type_to_snow_type(
             )
         return TimestampType(timezone), False
 
+    if tp == object:
```

Comment:
where do you need this? Maybe implement this in pandas_lit for now. This mapping may need more discussion I believe.

Reply:
This is needed to support 'object' type hint, which is where this is called, the test test_Apply_axis_0 with parameter [[[1.1, 2.2], [3, np.nan]], lambda x: "0" if x.sum() > 1 else 0, "object"]) which essientally creates a type hint -> object for the apply function.

It is called via

check_return_variant_and_get_return_Type (apply_utils)
--> get_types_from_type_hints (udf_utils)
-->  calls python_type_to_snow_type (type_utils)

The type hint will fail to be recognized with a TypeError("invalid type <class 'object'>") without this check.

@sfc-gh-jdu looks like your familiar with this code, can you check this is best way to support 'object' type hint?

Reply:
Why user want to create a type hint for object?

Reply:
It looks like the original frame/test_apply.py test (for axis=1) to check type hints (including 'object') was added by https://snowflakecomputing.atlassian.net/browse/SNOW-1042200 (@sfc-gh-lspiegelberg) as a performance improvement, he may have more insight why type hint for variant would be useful.  In this case we're building on top of existing udtf apply code (specifically groupby apply) which do check these type hints if they are provided as found here.

Reply:
I was wondering why it worked for axis=1 cases but not now and then I found this https://github.com/snowflakedb/snowpark-python/blob/f6438409cadd28bfc947dc6ce164f36718b36fb6/src/snowflake/snowpark/modin/plugin/_internal/apply_utils.py#L813
Can we instead reuse `deduce_return_type_from_function` for `check_return_variant_and_get_return_Type`?

Reply:
Updated